main/src/cgeo/geocaching/connector/su/SuConnector.java

```diff
@@ -0,0 +1,249 @@
+        final StatusCode status = suLogin.login();
+
+        if (ConnectorFactory.showLoginToast && handler != null) {
+            handler.sendMessage(handler.obtainMessage(0, status));
+            ConnectorFactory.showLoginToast = false;
+
+            // invoke settings activity to insert login details
+            if (status == StatusCode.NO_LOGIN_INFO_STORED && fromActivity != null) {
+                SettingsActivity.openForScreen(R.string.preference_screen_su, fromActivity);
+            }
+        }
+        return status == StatusCode.NO_ERROR;
+    }
+
+    @Override
+    public String getUserName() {
+        return suLogin.getActualUserName();
+    }
+
+    @Override
+    public int getCachesFound() {
+        return suLogin.getActualCachesFound();
+    }
+
+    @Override
+    public String getLoginStatusString() {
+        return suLogin.getActualStatus();
+    }
+
+    @Override
+    public boolean isLoggedIn() {
+        return suLogin.isActualLoginStatus();
+    }
+
+    @Override
+    public Credentials getCredentials() {
+        return Settings.getCredentials(R.string.pref_suusername, R.string.pref_supassword);
+    }
+
+    @Override
+    public int getUsernamePreferenceKey() {
+        return R.string.pref_suusername;
+    }
+
+    @Override
+    public int getPasswordPreferenceKey() {
+        return R.string.pref_supassword;
+    }
+
+    @Override
+    public int getAvatarPreferenceKey() {
+        return R.string.pref_su_avatar;
+    }
+
+    @Override
+    @NonNull
+    public String getName() {
+        return "Geocaching.su";
+    }
+
+    @Override
+    @NonNull
+    public String getNameAbbreviated() {
+        return "GCSU";
+    }
+
+    @Override
+    @NonNull
+    public String getCacheUrl(@NonNull final Geocache cache) {
+        return getCacheUrlPrefix() + "&cid=" + cache.getCacheId();
+    }
+
+    @Override
+    public boolean getHttps() {
+        return false;
+    }
+
+    @Override
+    @NonNull
+    public String getHost() {
+        return "www.geocaching.su";
+    }
+
+    @Override
+    public boolean isOwner(@NonNull final Geocache cache) {
+        return StringUtils.isNotEmpty(getUserName()) && StringUtils.equals(cache.getOwnerDisplayName(), getUserName());
+    }
+
+    @Override
+    public boolean canHandle(@NonNull final String geocode) {
+        return StringUtils.startsWithAny(StringUtils.upperCase(geocode), PREFIX_GENERAL, PREFIX_TRADITIONAL, PREFIX_MULTISTEP_VIRTUAL, PREFIX_VIRTUAL, PREFIX_MULTISTEP, PREFIX_EVENT, PREFIX_CONTEST, PREFIX_MYSTERY, PREFIX_MYSTERY_VIRTUAL) && isNumericId(geocode.substring(2));
+    }
+
+    @Override
+    @NonNull
+    protected String getCacheUrlPrefix() {
+        return getHostUrl() + "/?pn=101";
+    }
+
+    @Override
+    public boolean isActive() {
+        return Settings.isSUConnectorActive();
+    }
+
+    @Override
+    public SearchResult searchByCenter(@NonNull final Geopoint center) {
+        return new SearchResult();
```

Comment:
no search by center? that will affect nearby and other search contexts which lead to list results.

Reply:
There was no such method in new API. Let me try to re-implement it